shared-spring/src/main/java/tds/shared/spring/interceptors/RestTemplateLoggingInterceptor.java

```diff
@@ -34,18 +35,25 @@ public RestTemplateLoggingInterceptor(final ObjectMapper objectMapper) {
     public ClientHttpResponse intercept(final HttpRequest request,
                                         final byte[] body,
                                         final ClientHttpRequestExecution clientHttpRequestExecution) throws IOException {
-        final UUID traceId = UUID.randomUUID();
+
+        // get tracer id that used to associate all rest calls that are a child of this request
+        final String traceId = Optional.fromNullable(MDC.get(TRACER_ID_HEADER)).or(UUID.randomUUID().toString());
         logRequest(request, body, traceId);
-        ClientHttpResponse response = clientHttpRequestExecution.execute(request, body);
-        logResponse(response, traceId);
-        return response;
+        try {
+            ClientHttpResponse response = clientHttpRequestExecution.execute(request, body);
+            logResponse(response, traceId);
+            return response;
+        } catch (IOException e) {
+            log.error(String.format("Exception occurred while executing rest request. METHOD: %s URI: %s", request.getMethod(), request.getURI()), e);
+            throw e;
```

Comment:
This interceptor is not meant to change the control flow of the application.  The application should behave the same with or without the interceptor.  The logic defined before this commit was correct.  The problem was there were no logs for rest template exceptions in Student.  For instance, when the micro-services were not running.  The UI would display a meaningful error message, but there was no log of the problem.   Proctor is logging rest template exceptions sufficiently.